doc/source/integer_na.rst

```diff
@@ -0,0 +1,87 @@
+.. currentmodule:: pandas
+
+.. ipython:: python
+    :suppress:
+
+    import numpy as np
+    import pandas as pd
+
+ .. _integer_na:
+
+**************************
+Nullable Integer Data Type
+**************************
+
+.. versionadded:: 0.24.0
+
+In :ref:`missing_data`, we saw that pandas primarily uses ``NaN`` to represent
+missing data. Because ``NaN`` is a float, this forces an array of integers with
+any missing values to become floating point. In some cases, this may not matter
+much. But if your integer column is, say, and identifier, casting to float can
+be problematic.
```

Comment:
Pretty niche case, but could also make reference to integers not representable in float64 space, e.g.
https://stackoverflow.com/questions/3793838/which-is-the-first-integer-that-an-ieee-754-float-is-incapable-of-representing-e